server/api/resolvers/requestResolvers.ts

```diff
@@ -0,0 +1,123 @@
+        return Request.find().exec()
+    },
+    requestsFilter: async (_, { filter, options }, ___): Promise<Array<RequestInterface>> => {
+        return Request.find().exec()
+    },
+}
+
+const requestMutationResolvers = {
+    createRequest: async (_, { request }, { authenticateUser }): Promise<RequestInterface> => {
+        return authenticateUser().then(async () => { 
+            return sessionHandler(async (session) => {
+                const newRequestObject = new Request({...request})
+                const newRequest = await newRequestObject.save({ session: session })
+                
+                const requestType = await RequestType.findById(newRequest.requestType).session(session)
+                requestType.requests.push(requestEmbeddingFromRequest(newRequest))
+                await requestType.save({ session: session })
+
+                return newRequest
+            })
+        })
+    },
+    updateRequest: async (_, { request }, { authenticateUser }): Promise<RequestInterface> => {
+        return authenticateUser().then(async () => {
+            return sessionHandler(async (session) => {
+                const oldRequest = await Request.findById(request._id).session(session)
+                const modifiedRequestObject = new Request({...oldRequest, ...request})
+                const newRequest = await modifiedRequestObject.save({ session: session })
+
+                if (oldRequest.requestType !== newRequest.requestType) {
+                    swapRequestTypeForRequest(newRequest, oldRequest.requestType, newRequest.requestType, session)
+                }
+
+                return newRequest
+            })
+        })
+    },
+    deleteRequest: async (_, { _id }, { authenticateUser }): Promise<RequestInterface> => {
+        return authenticateUser().then(async () => {
+            const request = await Request.findById(_id)
+            request.deletedAt = new Date()
+            return request.save()
+        })
+    },
+    fulfillRequest: async (_, { _id }, { authenticateUser }): Promise<RequestInterface> => {
+        return authenticateUser().then(async () => {
+            const request = await Request.findById(_id)
+            request.fulfilledAt = new Date()
+            return request.save()
+        })
+    },
+    unfulfillRequest: async (_, { _id }, { authenticateUser }): Promise<RequestInterface> => {
+        return authenticateUser().then(async () => {
+            const request = await Request.findById(_id)
+            request.fulfilledAt = undefined
+            return request.save()
+        })
+    },
+    changeRequestTypeForRequest: async (_, { request, requestType }, { authenticateUser }): Promise<RequestInterface> => {
+        return authenticateUser().then(async () => {
+            return sessionHandler(async (session) => {
+                const modifiedRequestObject = await Request.findById(request).session(session)
```

Comment:
why are we passing request into the findById. should it be request._id?

Reply:
request's type is ID. Should the name be changed?

Reply:
Modified it to "requestId" and "requestTypeId"